cpp/util/ManagedArray.h

```diff
@@ -205,7 +219,13 @@ template<typename T> class ManagedArray
     {
         size_t cur_prod = 1;
         size_t idx = 0;
-        for (int i = indices.size() - 1; i >= 0; --i)
+        // In getting the linear bin, we must iterate over bins in reverse
+        // order to build up the value of cur_prod because each subsequent axis
+        // contributes less according to row-major ordering. The condition in
+        // the for loop exploits the fact that i-- returns the pre-decrement
+        // value, so when i first reaches 0 the comparison will be 1 > 0 and
+        // ensure that the 0 case runs.
+        for (unsigned int i = indices.size(); i-- > 0;)
```

Comment:
This isn't as readable as the previous code, especially for new developers. See other comment above.